### ISSUE TITLE
Custom log driver not available in task definition configuration

### ISSUE THREAD

**Issue by kkost** [aws/amazon-ecs-agent#2286]
### Summary
tldr; I'm trying to configure Grafana's [Loki](https://github.com/grafana/loki/tree/master/cmd/docker-driver) log driver in my task definition.

### Description
I have the log driver plugin installed on my ECS cluster instances, and have added `loki` to the `ECS_AVAILABLE_LOGGING_DRIVERS=...` line in /etc/ecs/ecs.config.

If I configure the logConfiguration section programatically (via Terraform my this case) as such:

```
...
    "logConfiguration": {
      "logDriver": "loki",
      "options": {
        "loki-url": "http://some-loki-endpoint/loki/api/v1/push",
        "loki-batch-size": "400"
      }
    }
...
```
the update is accepted without error. But the task never starts and no error is provided in the service's Events section.

If I view the task def through the console, my "loki" logConfiguration is not shown ("\<default\>" selected). Though it does show the config I set in the "Configure via JSON" output.

Possible workaround is to NOT configure the logConfiguration section (it *should* use the docker daemon's default in /etc/docker/daemon.json, which I'd set to use loki myself, if it's not specified). Or start using Grafana's [Promtail](https://github.com/grafana/loki/tree/master/docs/clients/promtail) to push logs.

### Expected Behavior
Ability to type in a custom log driver if it isn't in the list. Or to have "loki" added to the list.

### Observed Behavior
Task doesn't start

### Environment Details
n/a

### Supporting Log Snippets
n/a

**Comment by suneyz** [comment]
Hi @kkost , thanks for reaching out. We don’t currently support loki custom logging drivers as of now. 

We have an existing feature request https://github.com/aws/containers-roadmap/issues/435 on our container roadmap to support arbitrary custom log driver. Closing issue as duplicate.
